hugr-model/src/v0/text/parse.rs

```diff
@@ -627,6 +751,7 @@ impl<'a> ParseContext<'a> {
 
         for pair in filter_rule(pairs, Rule::param) {
             let param = pair.into_inner().next().unwrap();
+            let param_span = param.as_span();
```

Comment:
might as well only call this in the error closure?

Reply:
It'll probably be optimised to be that, but I agree that it might be nicer to be explicit about it and put it in the closure.

Reply:
Ah, I remember now why I did that. The subsequent code calls `.into_inner` on `param`, which consumes `param`. Therefore `param` is no longer available when the error is constructed. So either we'd have to clone `param` to get the error span from it later or do it like it is now. The clone would have to `Rc` ref count increases, so it's probably no different performance wise. Also `pest` is rather slow anyway and parsing isn't on the hot path (currently).